UpdatePaycoinNode.sh

```diff
@@ -1,16 +1,48 @@
 #!/bin/bash
+version=$(curl -s https://api.github.com/repos/PaycoinFoundation/Paycoin/releases/latest | grep 'tag_' | cut -d\" -f4 | sed 's/\v//g')
+latest=$(curl -s https://api.github.com/repos/PaycoinFoundation/Paycoin/releases/latest | grep 'browser_' | cut -d\" -f4 | grep 'linux64.zip')
 echo "### Changing to paycoind directory"
 cd ~
-echo "### Stopping Paycoin Server"
-./paycoind stop
-echo "### Changing to home directory"
-cd ~
-echo "### Downloading Paycoin Core 0.3.3.0"
-wget https://github.com/PaycoinFoundation/paycoin/releases/download/v0.3.3.0/linux64.zip
-echo "### Installing Paycoin Core 0.3.3.0"
+
+xpy_version=$(./paycoind getinfo 2> /dev/null | grep '"version"' | cut -d\" -f4 | sed 's/\v//g' >/dev/null)
+# Check we are running the same version; otherwise; update
+
+if [[ "${xpy_version}" == "" ]]
+then
+    echo "### Paycoin Server Not Running, Unable to Check Version"
+    echo "### The latest is available version is v${version}"
+else 
+    if [[ "${version}" == "${xpy_version}" ]]
+    then
+        echo "### You are already running the latest version: v${version}"
+        echo "### Exiting updater script."
+        exit 1
+    else
+        echo "### You are running Paycoin Core v#{xpy_version}"
+        echo "### The latest is available version is v${version}"
```

Comment:
That is definitely an error on my part. Must have missed it (the brain is funny
like that)
I'll patch it and push it back up

On Fri, Nov 20, 2015 at 8:32 a.m., Mitchell Cash <notifications@github.com>
wrote:
In UpdatePaycoinNode.sh
[https://github.com/mitchellcash/PaycoinAutoNode/pull/13#discussion_r45445541] :

> +xpy_version=$(./paycoind getinfo 2> /dev/null | grep '"version"' | cut -d\" -f4 | sed 's/\v//g' >/dev/null)
> +# Check we are running the same version; otherwise; update
> +
> +if [[ "${xpy_version}" == "" ]]
> +then
> +    echo "### Paycoin Server Not Running, Unable to Check Version"
> +    echo "### The latest is available version is v${version}"
> +else 
> +    if [[ "${version}" == "${xpy_version}" ]]
> +    then
> +        echo "### You are already running the latest version: v${version}"
> +        echo "### Exiting updater script."
> +        exit 1
> +    else
> +        echo "### You are running Paycoin Core v#{xpy_version}"
> +        echo "### The latest is available version is v${version}"


Not yet it hasn't :P

It's got one too many 'is', something like 'The latest available version is
v${version}' works better.

—
Reply to this email directly or view it on GitHub
[https://github.com/mitchellcash/PaycoinAutoNode/pull/13/files#r45445541] .[https://github.com/notifications/beacon/ABeWHZgFKcEcjeKAQCFpymP5mNixHe1pks5pHtIYgaJpZM4GkBrN.gif]